great_expectations/rule_based_profiler/parameter_container.py

```diff
@@ -230,18 +230,37 @@ def to_json_dict(self) -> dict:
 def convert_dictionary_to_parameter_node(
     source: Union[T, dict]
 ) -> Union[T, ParameterNode]:
-    if not isinstance(source, dict):
-        return source
+    if isinstance(source, dict):
+        return _convert_dictionary_to_parameter_node(source=ParameterNode(source))
 
-    return _convert_dictionary_to_parameter_node(source)
+    # Must allow for non-dictionary source types, since their internal nested structures may contain dictionaries.
```

Comment:
Same comment here regarding method naming

Reply:
@NathanFarmer Yes -- this is the third of three.  Thanks.